src/main/java/group1/mips_simulator/components/Computer.java

```diff
@@ -113,26 +114,22 @@ public boolean executeInstruction(Instruction instruction) {
             default ->
                     throw new IllegalArgumentException("Unknown instruction op code name: " + instruction.opCode.name +
                             "\nInstruction: " + instruction);
-        }
-        // After every instruction then we increment the Program Counter
-        // EXCEPT if the instruction is a transfer instruction. Transfer instructions
-        // are responsible for setting the PC themselves.
-        if (!instruction.isTransferInstruction()) {
-            this.incrementPC();
-        }
+        };
+
+        // Set the PC to whatever the instruction results dictates
+        this.cpu.regfile.getPC().write(executionResult.newPC);
 
-        // Return if the instruction is HLT then return false to signal that
-        // the program should shut down.
-        return !instruction.isHaltInstruction();
+        // Return if the instruction is HLT, or an error has occurred, or whatever other reason
+        // to signal the stop of the computer
+        return executionResult.shouldHaltComputer;
     }
 
-    /**
-     * Update the program counter register by incrementing it to the
-     * next position (+1)
-     */
```

Comment:
very sexy code change, just would recommend keeping the comments (/updating them)